source/core/array.h

```diff
@@ -6,138 +6,130 @@
 
 namespace Slang
 {
-	template<typename T, int size>
+	template<typename T, int COUNT>
```

Comment:
Minor nit: I'm not sure what our naming convention should be for non-type template parameters, but whatever it is I'm unconvinced on ALL CAPS. :)